src/authentication.ts

```diff
@@ -48,6 +49,11 @@ class CustomisedAuthenticationService extends AuthenticationService {
 }
 
 class HashedPasswordVerifier extends LocalStrategy {
+  app: ImpressoApplication
+  constructor(app: ImpressoApplication) {
+    super()
+    this.app = app
+  }
   comparePassword(user: User, password: string) {
     return new Promise((resolve, reject) => {
```

Comment:
It's a great opportunity to convert the then/catch promise chain to an async function. Can you do it here?